pkg/apis/extensions/v1beta1/types.go

```diff
@@ -217,6 +217,26 @@ type DeploymentSpec struct {
 	// Indicates that the deployment is paused and will not be processed by the
 	// deployment controller.
 	Paused bool `json:"paused,omitempty"`
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	RollbackTo *RollbackConfig `json:"rollbackTo,omitempty"`
+}
+
+// DeploymentRollback stores the information required to rollback a deployment.
+type DeploymentRollback struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Required: This must match the Name of a deployment.
+	Name string `json:"name"`
+	// Required: This must match the Namespace of a deployment.
+	Namespace string `json:"namespace"`
+	// The annotations to be updated to a deployment
+	UpdatedAnnotations map[string]string `json:"updatedAnnotations,omitempty"`
+	// The config of this deployment rollback.
+	RollbackTo *RollbackConfig `json:"rollbackTo"`
```

Comment:
Is there any reason we'd make RollbackTo optional? If not, it shouldn't be a pointer. If so, it needs to be tagged with omitempty.